geomesa-features/geomesa-feature-avro/src/main/scala/org/locationtech/geomesa/features/avro/AvroSimpleFeatureTypeParser.scala

```diff
@@ -0,0 +1,458 @@
+/***********************************************************************
+ * Copyright (c) 2013-2021 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.features.avro
+
+import org.apache.avro.Schema
+import org.apache.avro.generic.GenericRecord
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder
+import org.joda.time.format.ISODateTimeFormat
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.locationtech.geomesa.utils.text.{WKBUtils, WKTUtils}
+import org.locationtech.jts.geom.{Geometry, GeometryCollection, LineString, MultiLineString, MultiPoint, MultiPolygon, Point, Polygon}
+import org.opengis.feature.simple.SimpleFeatureType
+
+import java.nio.ByteBuffer
+import java.util.{Date, Locale}
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+import scala.reflect.{ClassTag, classTag}
+
+object AvroSimpleFeatureTypeParser {
+  /**
+   * Convert an Avro [[Schema]] into a [[SimpleFeatureType]].
+   */
+  def schemaToSft(schema: Schema): SimpleFeatureType = {
+    val builder = new SimpleFeatureTypeBuilder
+    builder.setName(schema.getName)
+
+    // any additional top-level props on the schema go in the sft user data
+    val userData = schema.getProps
+
+    schema.getFields.foreach { field =>
+      val fieldName = field.name()
+      val metadata = parseMetadata(field)
+
+      metadata.properties match {
+        case GeomProperties(_, geomType, default, srid) =>
+          builder.add(fieldName, geomType)
+          // if more than one field is set as the default geometry, the last one becomes the default
+          if (default) { builder.setDefaultGeometry(fieldName) }
+          srid.map(builder.get(fieldName).getUserData.put(SimpleFeatureTypes.AttributeOptions.OptSrid, _))
+        case DateProperties(_, default) =>
+          builder.add(fieldName, classOf[Date])
+          // if more than one field is set as the default date, the last one becomes the default
+          if (default) { userData.put(SimpleFeatureTypes.Configs.DefaultDtgField, fieldName) }
+        case VisibilityProperties =>
+          // if there is more than one visibilities field, the last one is used
```

Comment:
I'd opine that it should be an error if there are multiple vis fields.

I mean, I hope no one does that, but I wouldn't want to debug it otherwise....:)

Reply:
Fixed